src/node.ts

```diff
@@ -385,46 +387,58 @@ export abstract class BaseNode<
    * @returns a mapping of name to the node visible in this node's scope.
    */
   public getLexicalScope(): Map<string, BindingDecl> {
-    return new Map(getLexicalScope(this as unknown as FunctionlessNode));
+    return new Map(
+      getLexicalScope(this as unknown as FunctionlessNode, "scope")
+    );
 
-    function getLexicalScope(node: FunctionlessNode | undefined): Binding[] {
+    function getLexicalScope(
+      node: FunctionlessNode | undefined,
+      kind: "scope" | "sibling"
+    ): Binding[] {
       if (node === undefined) {
         return [];
       }
       return getLexicalScope(
-        node.nodeKind === "Stmt" && node.prev ? node.prev : node.parent
-      ).concat(getNames(node));
+        node.nodeKind === "Stmt" && node.prev ? node.prev : node.parent,
+        node.nodeKind === "Stmt" && node.prev ? "sibling" : "scope"
+      ).concat(getNames(node, kind));
     }
 
-    function getNames(node: FunctionlessNode | undefined): Binding[] {
+    function getNames(
+      node: FunctionlessNode | undefined,
+      kind: "scope" | "sibling"
+    ): Binding[] {
       if (node === undefined) {
         return [];
+      } else if (isParameterDecl(node)) {
+        return isIdentifier(node.name)
+          ? [[node.name.name, node]]
+          : getNames(node.name, kind);
+      } else if (isVariableDeclList(node)) {
+        return node.decls.flatMap((d) => getNames(d, kind));
       } else if (isVariableStmt(node)) {
-        return node.declList.decls.flatMap(getNames);
+        return getNames(node.declList, kind);
       } else if (isVariableDecl(node)) {
         if (isBindingPattern(node.name)) {
-          return getNames(node.name);
+          return getNames(node.name, kind);
         }
         return [[node.name.name, node]];
       } else if (isBindingElem(node)) {
         if (isIdentifier(node.name)) {
           return [[node.name.name, node]];
         }
-        return getNames(node.name);
+        return getNames(node.name, kind);
       } else if (isBindingPattern(node)) {
-        return node.bindings.flatMap((b) => getNames(b));
+        return node.bindings.flatMap((b) => getNames(b, kind));
       } else if (isFunctionLike(node)) {
-        return node.parameters.flatMap((param) =>
-          isIdentifier(param.name)
-            ? [[param.name.name, param]]
-            : getNames(param.name)
-        );
-      } else if (isForInStmt(node) || isForOfStmt(node)) {
-        return getNames(node.initializer);
-      } else if (isForStmt(node)) {
-        return getNames(node.initializer);
+        if (kind === "sibling") return [];
```

Comment:
Why? I don't understand what `sibling` is modeling

Reply:
This code was returning names from scopes like function and for even after their scope was complete.

This is an attempt to inform the nodes types when to output their scoped names.

```
1 for(const a in []) { // scope
2  const b = ""; // sibling
3   for(const a in []){ ... } //sibling
4   a // node.getLexicalScope() -> { a: 1 }
5 }
```

I'll do another pass at this, but was happy to get it to behave as I intended.